### PR TITLE
GCNConv self loops

### PR DESCRIPTION
**Pull request by nkanak**
## Proposed changes

Enable self-loops for the GCNConv layer.

## Checklist

Put an `x` in the boxes that apply.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation

### CODE REVIEW

File: mlx_graphs/nn/conv/gcn_conv.py [hunk 1]
```diff
@@ -14,19 +14,22 @@ class GCNConv(MessagePassing):
...
         out_features_dim: int,
         bias: bool = True,
+        add_self_loops: bool = False,
         **kwargs,
     ):
         kwargs.setdefault("aggr", "add")
         super(GCNConv, self).__init__(**kwargs)
 
         self.linear = nn.Linear(node_features_dim, out_features_dim, bias)
+        self.add_self_loops = add_self_loops
```
Comment (francescofarina) [inline mlx_graphs/nn/conv/gcn_conv.py:32]: Shall we make this private?

**Review by francescofarina — CHANGES_REQUESTED** [review]

File: mlx_graphs/nn/conv/gcn_conv.py (hunk 1 above)
Comment (francescofarina) [reply mlx_graphs/nn/conv/gcn_conv.py:32]:
```suggestion
        self._add_self_loops = add_self_loops
```

Comment (nkanak) [reply mlx_graphs/nn/conv/gcn_conv.py]: Ready!

**Review by francescofarina — APPROVED** [review]
Looks good 🚢

**Pull request merged**